src/Core/Components/Rating/FluentRating.razor.cs

```diff
@@ -0,0 +1,88 @@
+using Microsoft.AspNetCore.Components;
+using Microsoft.AspNetCore.Components.Web;
+using System.Diagnostics.CodeAnalysis;
+using System.Globalization;
+
+namespace Microsoft.FluentUI.AspNetCore.Components;
+
+public partial class FluentRating : FluentInputBase<int>
+{
+    private int? _mouseOverValue;
+
+    public FluentRating() => Id = Identifier.NewId();
+
+    /// <summary>
+    /// Gets or sets the maximum value.
+    /// </summary>
+    [Parameter] public int MaxValue { get; set; } = 10;
+
+
+    /// <summary>
+    /// Gets or sets the icon drawing and fill color. 
+    /// Value comes from the <see cref="AspNetCore.Components.Color"/> enumeration. Defaults to Accent.
+    /// </summary>
+    [Parameter] public Color? IconColor { get; set; } 
+
+    /// <summary>
+    /// Selected or hovered icon.
+    /// </summary>
+    [Parameter] public Icon FullIcon { get; set; } = new CoreIcons.Filled.Size20.Star();
+
+    /// <summary>
+    /// Non-selected item icon.
+    /// </summary>
+    [Parameter] public Icon EmptyIcon { get; set; } = new CoreIcons.Regular.Size20.Star();
+
+    private Icon GetIcon(int index) => index <= (_mouseOverValue ?? Value) ? FullIcon : EmptyIcon;
+
+    protected override bool TryParseValueFromString(string? value, [MaybeNullWhen(false)] out int result, [NotNullWhen(false)] out string?
+        validationErrorMessage)
+    {
+        if (BindConverter.TryConvertTo(value, CultureInfo.InvariantCulture, out result))
+        {
+            validationErrorMessage = null;
+            return true;
+        }
+        else
+        {
+            validationErrorMessage = string.Format(CultureInfo.InvariantCulture,
+                                                   "The {0} field must be a number.",
+                                                   FieldBound ? FieldIdentifier.FieldName : UnknownBoundField);
+            return false;
+        }
+    }
+
+    protected internal async Task HandleKeyDownAsync(KeyboardEventArgs keyboardEventArgs)
```

Comment:
Keyboard usage

Reply:
Are you sure about the logic? 
Shift+ArrowRight is now MaxValue (say 10) - Value (say 4) so with `SetCurrentValueAsync` you set Value to 6?

Also, please make ArrowDown do the same as ArrowLeft and make ArrowUp do same as ArrowRight (and shift + down/up as well of course)

Reply:
I agree, I would find it cleaner to use a classic `switch` and add `Value++` or `Value = MaxValue` in the cases.